### ISSUE TITLE
Update to deck.gl v8.0.0

### ISSUE THREAD

**Issue by blinky3713** [f-o-a-m/purescript-deck-gl#12]
We are currently depending v5.3.0. As far as I can tell, the [recently updated](https://github.com/f-o-a-m/purescript-react-map-gl/tree/v0.9.0) (and hopefully working) `purescript-react-map-gl` has compatible dependencies, as deck.gl` is requiring 

```json
    "react": "^16.2.0",
    "react-map-gl": "^5.1.0"
```

The current version of master has already pulled in this dependency and updated everything except for deck.gl itself.

Naively bumping the version to `6.*` or `7.*` completely breaks the icon example. Naively updating to `8.*` breaks it, but at least something non-trivial loads. deck.gl has a [changelog](https://github.com/visgl/deck.gl/blob/master/CHANGELOG.md) as well as an [upgrade path](https://github.com/visgl/deck.gl/blob/master/docs/upgrade-guide.md)

Right now we only have a test for the `icon` layer. I think it's sufficient to get this working again to close this issue. However as a separate issue, we would also like to add another example to the `examples` directory here to further prove that it works. The [trip-layer](https://github.com/visgl/deck.gl/blob/master/docs/api-reference/geo-layers/trips-layer.md) was added in version `7` as a stable layer, and has an updated API in version `8`. If we can make a small app that mimics [their trip layer app](https://github.com/visgl/deck.gl/tree/master/examples/website/trips) then I would be more convinced that we understand the updated api.

**Comment by blinky3713** [comment]
For starters, when I try to naively try to upgrade to `^v6.0.0` (which resolves to v6.4.10 for me), and build the icon app, I get the following error:

![image](https://user-images.githubusercontent.com/31280145/90281863-381e8500-de22-11ea-9a82-5b7ba8e954d5.png)

So then I print the `defaultIconProps` being passed to `makeIconLayer` and I can see that there are multiple undefined fields:

![image](https://user-images.githubusercontent.com/31280145/90281968-69975080-de22-11ea-8bf6-34c48e33f1b8.png)

Which leads me to believe that the `DeckGL.Layer.defaultProps` are not being merged with the layer specific props anymore (via DeckGL.Layer.defaultProps`. So then the definition of `defaultIconProps` to 

```javascript
exports.defaultIconProps = Immutable.merge(DeckGL.Layer.defaultProps, DeckGL.IconLayer.defaultProps);
```

I can see that the fields are no longer undefined but that there's a new error which I haven't been successful in debugging

![image](https://user-images.githubusercontent.com/31280145/90282319-ff32e000-de22-11ea-925f-9609448ed3de.png)

**Comment by srghma** [comment]
done, 1 sec will push

![2020-09-29-18:28:36-screenshot](https://user-images.githubusercontent.com/7573215/94579619-b4a9dd80-0281-11eb-8694-6510eb984f30.png)
